src/compiler/scala/tools/nsc/util/RunnableInPhase.scala

```diff
@@ -0,0 +1,34 @@
+package scala.tools.nsc.util
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Awaitable, Future}
+import scala.tools.nsc.{Global, Phase}
+object RunnableInPhase {
+  val idGen = new AtomicInteger
+}
+/**
+  * a wrapper to allow Runnables to be associated to a phase. Primarily useful for profiling
```

Comment:
see https://github.com/scala/scala/pull/5815 for use - this enables the profiler to be aware of the background thread
profiler is in https://github.com/scala/scala/pull/5760 (early version) and current working version in https://github.com/rorygraves/scalac_perf/tree/2.12.x_profile_baseline

when used with -Yprofile-enable:true

this is a pre-req for the profiler and not used directly in this PR